core/src/main/java/io/grpc/internal/DelayedClientTransport2.java

```diff
@@ -85,13 +87,50 @@
   @Nullable
   private SubchannelPicker lastPicker;
 
-  DelayedClientTransport2(Executor streamCreationExecutor) {
-    this.streamCreationExecutor = streamCreationExecutor;
+  @GuardedBy("lock")
+  private long lastPickerVersion;
+
+  /**
+   * Creates a new delayed transport.
+   *
+   * @param defaultAppExecutor pending streams will create real streams and run bufferred operations
+   *        in an application executor, which will be this executor, unless there is on provided in
+   *        {@link CallOptions}.
+   * @param channelExecutor all listener callbacks of the delayed transport will be run from this
+   *        ChannelExecutor.
+   */
+  DelayedClientTransport2(Executor defaultAppExecutor, ChannelExecutor channelExecutor) {
+    this.defaultAppExecutor = defaultAppExecutor;
+    this.channelExecutor = channelExecutor;
   }
 
   @Override
-  public final Runnable start(Listener listener) {
-    this.listener = Preconditions.checkNotNull(listener, "listener");
+  public final Runnable start(final Listener listener) {
+    this.reportTransportInUse = new Runnable() {
+        @Override
+        public void run() {
+          listener.transportInUse(true);
+        }
+      };
+    this.reportTransportNotInUse = new Runnable() {
+        @Override
+        public void run() {
+          listener.transportInUse(false);
+        }
+      };
+    this.reportTransportShutdown = new Runnable() {
```

Comment:
Just for consistency with other callbacks.  And slightly easier for me to make sure that all callbacks are run from channelExecutor.